File: backend/src/modules/carrierroute/carrierroute.controller.ts
import { Controller, Get, Post, Body, Delete, Param, Query, ParseIntPipe } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { CarrierrouteService } from './carrierroute.service';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('运营商路由')
@Controller('api/carrierroute')
export class CarrierrouteController {
  constructor(private readonly carrierrouteService: CarrierrouteService) {}

  // Carriers
  @Get('carriers')
  @ApiOperation({ summary: '获取运营商列表' })
  async findAllCarriers() {
    return ApiResponseDto.success(await this.carrierrouteService.findAllCarriers());
  }

  @Post('carriers')
  async createCarrier(@Body() body: { carrier: string }) {
    return ApiResponseDto.success(await this.carrierrouteService.createCarrier(body.carrier), '创建成功');
  }

  @Delete('carriers/:id')
  async deleteCarrier(@Param('id', ParseIntPipe) id: number) {
    await this.carrierrouteService.deleteCarrier(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Domains
  @Get('domains')
  @ApiOperation({ summary: '获取域列表' })
  async findAllDomains() {
    return ApiResponseDto.success(await this.carrierrouteService.findAllDomains());
  }

  @Post('domains')
  async createDomain(@Body() body: { domain: string }) {
    return ApiResponseDto.success(await this.carrierrouteService.createDomain(body.domain), '创建成功');
  }

  @Delete('domains/:id')
  async deleteDomain(@Param('id', ParseIntPipe) id: number) {
    await this.carrierrouteService.deleteDomain(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Failure Routes
  @Get('failure-routes')
  @ApiOperation({ summary: '获取失败路由列表' })
  async findAllFailureRoutes(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.carrierrouteService.findAllFailureRoutes(dto));
  }

  @Post('failure-routes')
  async createFailureRoute(@Body() data: any) {
    return ApiResponseDto.success(await this.carrierrouteService.createFailureRoute(data), '创建成功');
  }

  @Delete('failure-routes/:id')
  async deleteFailureRoute(@Param('id', ParseIntPipe) id: number) {
    await this.carrierrouteService.deleteFailureRoute(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // RPC
  @Post('reload')
  @ApiOperation({ summary: '重载路由' })
  async reload() {
    await this.carrierrouteService.reload();
    return ApiResponseDto.success(null, '重载成功');
  }

  @Get('dump')
  @ApiOperation({ summary: '导出路由' })
  async dumpRoutes() {
    return ApiResponseDto.success(await this.carrierrouteService.dumpRoutes());
  }
}

File: kam-admin-module-routing/src/main/java/io/kamailio/admin/modules/carrierroute/controller/CarrierrouteController.java
package io.kamailio.admin.modules.carrierroute.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.carrierroute.entity.CarrierFailureRoute;
import io.kamailio.admin.modules.carrierroute.entity.CarrierName;
import io.kamailio.admin.modules.carrierroute.entity.DomainName;
import io.kamailio.admin.modules.carrierroute.service.CarrierrouteService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@Tag(name = "运营商路由")
@RestController
@RequestMapping("/api/carrierroute")
public class CarrierrouteController {
    private final CarrierrouteService service;

    public CarrierrouteController(CarrierrouteService service) { this.service = service; }

    @GetMapping("/carriers")
    @Operation(summary = "获取运营商列表")
    public ResponseEntity<ApiResponse<List<CarrierName>>> findAllCarriers() {
        return ResponseEntity.ok(ApiResponse.success(service.findAllCarriers()));
    }

    @PostMapping("/carriers")
    public ResponseEntity<ApiResponse<CarrierName>> createCarrier(@RequestBody Map<String, String> body) {
        return ResponseEntity.ok(ApiResponse.success(service.createCarrier(body.get("carrier")), "创建成功"));
    }

    @DeleteMapping("/carriers/{id}")
    public ResponseEntity<ApiResponse<Void>> deleteCarrier(@PathVariable Integer id) {
        service.deleteCarrier(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/domains")
    @Operation(summary = "获取域列表")
    public ResponseEntity<ApiResponse<List<DomainName>>> findAllDomains() {
        return ResponseEntity.ok(ApiResponse.success(service.findAllDomains()));
    }

    @PostMapping("/domains")
    public ResponseEntity<ApiResponse<DomainName>> createDomain(@RequestBody Map<String, String> body) {
        return ResponseEntity.ok(ApiResponse.success(service.createDomain(body.get("domain")), "创建成功"));
    }

    @DeleteMapping("/domains/{id}")
    public ResponseEntity<ApiResponse<Void>> deleteDomain(@PathVariable Integer id) {
        service.deleteDomain(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/failure-routes")
    @Operation(summary = "获取失败路由列表")
    public ResponseEntity<ApiResponse<PaginatedResult<CarrierFailureRoute>>> findAllFailureRoutes(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllFailureRoutes(dto)));
    }

    @PostMapping("/failure-routes")
    public ResponseEntity<ApiResponse<CarrierFailureRoute>> createFailureRoute(@RequestBody CarrierFailureRoute data) {
        return ResponseEntity.ok(ApiResponse.success(service.createFailureRoute(data), "创建成功"));
    }

    @DeleteMapping("/failure-routes/{id}")
    public ResponseEntity<ApiResponse<Void>> deleteFailureRoute(@PathVariable Integer id) {
        service.deleteFailureRoute(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @PostMapping("/reload")
    @Operation(summary = "重载路由")
    public ResponseEntity<ApiResponse<Void>> reload() {
        service.reload();
        return ResponseEntity.ok(ApiResponse.success(null, "重载成功"));
    }

    @GetMapping("/dump")
    @Operation(summary = "导出路由")
    public ResponseEntity<ApiResponse<Object>> dumpRoutes() {
        return ResponseEntity.ok(ApiResponse.success(service.dumpRoutes()));
    }
}
